arwen/contexts/metering.go

```diff
@@ -236,6 +236,13 @@ func (context *meteringContext) GasUsedByContract() (uint64, uint64) {
 	return gasUsed, remainedFromForwarded
 }
 
+// GasUsedForExecution returns the actual gas used for execution for the contract which needs to be restored
+func (context *meteringContext) GasUsedForExecution() uint64 {
+	gasUsed, _ := context.GasUsedByContract()
+	gasUsed, _ = math.SubUint64(gasUsed, context.initialCost)
```

Comment:
Could overflow this subtraction in some edge case situation? SafeSub ?

Reply:
this means safeSub

Reply:
In method computeGasUsedInExecutionBeforeReset it is used core.SafeSubUint64 this is why I asked